File: Robotic Controllers/src/main/Entity.java
package main;

public class Entity {

    private int[] chromosome;
    private double fitness = -1;

    public Entity(int[] chromosome) {
        this.chromosome = chromosome;
    }

    Entity(int chromosomeLength) {
        this.chromosome = new int[chromosomeLength];

        // randomly generate a chromosome of length chromosomeLength
        for (int i = 0; i < chromosomeLength; i++) {
            if (Math.random() < 0.5) {
                this.setGene(i, 1);
            } else {
                this.setGene(i, 0);
            }
        }
    }

    public void setGene(int index, int value) {
        this.chromosome[index] = value;
    }

    public int[] getChromosome() {
        return this.chromosome;
    }

    public int chromosomeLength() {
        return this.chromosome.length;
    }

    public int getGene(int idx) {
        return this.chromosome[idx];
    }

    double getFitness() {
        return this.fitness;
    }

    public void setFitness(double fitness) {
        this.fitness = fitness;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (Integer i : chromosome
        ) {
            sb.append(i);
        }
        return sb.toString();
    }
}

File: models/entity.py
from typing import List

import numpy as np
from sklearn import datasets


class Entity:

	n_clusters: int
	chromosome: List[int]

	def __init__(self, n_clusters, X: np.ndarray):
		self.n_clusters = n_clusters
		self.chromosome = []
		self.dimensions = X.shape
		self.fitness = 0.0
		self.fill_chromosome()

	def fill_chromosome(self):
		self.chromosome = [0] * self.n_clusters * self.dimensions

	def random_center_choice(self, X: np.ndarray) -> np.ndarray:
		shape = X.shape
		rand_idx = np.random.permutation(shape[0])
		centers = X[rand_idx[:self.n_clusters]]
		return centers

	def calc_fitness(self):
		pass

	def get_fitness(self):
		return self.fitness

	def set_gene(self, idx: int, value: float):
		self.chromosome[idx] = value

	def get_gene(self, idx: int):
		return self.chromosome[idx]

	def chromosome_length(self):
		return len(self.chromosome)

	def set_fitness(self, fitness: float):
		self.fitness = fitness

	def __str__(self):
		return self.chromosome


def main():
	print("Hello World")
	iris = datasets.load_iris()
	X = iris.data
	y = iris.target
	print(y)

	entity = Entity(3, X)
	centers = entity.random_center_choice(X)
	print(entity)
	print("centers", centers)


if __name__ == "main":
	main()

